File: java/problem-10.java
// Project Euler - Problem 10
// Naveen Venkatesan
// March 22, 2014

public class Problem10 {

	public static boolean isPrime(int x) {
		
		if (x == 1) {
			return false;
		}
		
		if (x == 2) {
			return true;
		}

		for (int i = 2; i*i <= x; i++) {
			if (x % i == 0) {
				return false;
			}
		}

		return true;

	}

	public static void main(String[] args) {

		long sum = 0;

		for (int i = 1; i < 2000000; i++) {
			if (isPrime(i)) {
				sum += i;
			}
		}

		System.out.println("The sum of primes under two million is: " + sum);

	}

}

File: python/problem-10.py
### Problem 10 - Summation of Primes
###------------------------------------------------------
### The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
### Find the sum of all the primes below two million.

### Solution

import numpy as np

# Function to determine prime
def isPrime(n):
	if n == 1:
		return False
	if n == 2:
		return True
	for i in range(2, int(np.sqrt(n)) + 1):
		if n % i == 0:
			return False
	return True

sumOfPrimes = 0

for i in range(2000000):
	if isPrime(i):
		sumOfPrimes += i

print(sumOfPrimes)